src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -83,15 +92,28 @@ const reducer = (state: InitialState, action: ActionTypes) => {
     case WalletActions.SET_ADAPTERS:
       return { ...state, adapters: action.payload }
     case WalletActions.SET_WALLET:
-      return {
+      const stateData = {
         ...state,
         wallet: action.payload.wallet,
         walletInfo: {
           name: action?.payload?.name,
           icon: action?.payload?.icon,
-          deviceId: action?.payload?.deviceId
+          deviceId: action?.payload?.deviceId,
+          meta: { label: '', address: '' }
         }
       }
+
+      if (action.payload.meta && action.payload.meta.label)
+        stateData.walletInfo.meta.label = action.payload.meta.label
+
+      // ignoring the error here as this prop does not exist on HDWallet but it does
+      // still exist and it also gets populated
+      // @ts-ignore
+      if (action.payload.wallet.ethAddress)
+        // @ts-ignore
+        stateData.walletInfo.meta.address = action.payload.wallet.ethAddress
```

Comment:
Same comment about calculating this property in-place instead of mutating nested object properties

Reply:
fixed